oracle/price-feeder/oracle/oracle.go

```diff
@@ -116,19 +119,24 @@ func (o *Oracle) Start(ctx context.Context) error {
 		default:
 			o.logger.Debug().Msg("starting oracle tick")
 
-			startTime := time.Now()
+			// Wait for next block height to be available in the channel
+			nextBlockHeight := <-o.oracleClient.BlockHeightEvents
```

Comment:
what happens if we're processing height 100 and for some reason it takes too long such that we have height 101 and 102 in the channel? then the next height processed would be the old one, right? shouldn't we take EVERYTHING from the channel (if theres more than 1) and then process for latest block height?

Reply:
Right now each blocks is going to be processed in a separate thread, each broadcast takes 20ms-30ms to complete, so there's no possibility of being blocked at all. In the worse case where each broadcast take a few seconds to complete, the side effect is that we will end up creating more goroutines, where the num of goroutines= num of pending blocks within last few seconds

Reply:
this sounds good, I was just curious if there was some small chance of a backlog in the channel, but i think its reasonable that it would be processing fast enough